haystack/dataclasses/breakpoints.py

```diff
@@ -6,91 +6,65 @@
+    the breakpoint applies to all tools within the Agent component.
 
-    If tool_name is None, it means that the breakpoint is for every tool in the component.
-    Otherwise, it means that the breakpoint is for the tool with the given name.
+    :param tool_name: The name of the tool to target within the Agent component. If None, applies to all tools.
     """
 
     tool_name: Optional[str] = None
 
-    def __hash__(self):
-        return hash((self.component_name, self.visit_count, self.tool_name))
-
-    def __eq__(self, other):
-        if not isinstance(other, ToolBreakpoint):
-            return False
-        return super().__eq__(other) and self.tool_name == other.tool_name
-
     def __str__(self):
-        if self.tool_name:
-            return (
-                f"ToolBreakpoint(component_name={self.component_name}, visit_count={self.visit_count}, "
-                f"tool_name={self.tool_name})"
-            )
-        else:
-            return (
-                f"ToolBreakpoint(component_name={self.component_name}, visit_count={self.visit_count}, "
-                f"tool_name=ALL_TOOLS)"
-            )
-
-    def __repr__(self):
-        return self.__str__()
+        tool_str = f", tool_name={self.tool_name}" if self.tool_name else ", tool_name=ALL_TOOLS"
+        return f"ToolBreakpoint(component_name={self.component_name}, visit_count={self.visit_count}{tool_str})"
 
 
 @dataclass
 class AgentBreakpoint:
     """
-    A dataclass to hold a breakpoint that can be used to debug an Agent.
-    """
+    A dataclass representing a breakpoint tied to an Agent’s execution.
 
-    break_point: Union[Breakpoint, ToolBreakpoint]
-    agent_name: str = ""
+    This allows for debugging either a specific component (e.g., the chat generator) or a tool used by the agent.
+    It enforces constraints on which component names are valid for each breakpoint type.
 
-    def __init__(self, agent_name: str, break_point: Union[Breakpoint, ToolBreakpoint]):
-        if not isinstance(break_point, ToolBreakpoint) and break_point.component_name != "chat_generator":
-            raise ValueError(
-                "The break_point must be a Breakpoint that has the component_name "
-                "'chat_generator' or be a ToolBreakpoint."
-            )
+    :param agent_name: The name of the agent component in a pipeline where the breakpoint is set.
+    :param break_point: An instance of Breakpoint or ToolBreakpoint indicating where to break execution.
 
-        if not break_point:
-            raise ValueError("A Breakpoint must be provided.")
+    :raises ValueError: If the component_name is invalid for the given breakpoint type:
+        - Breakpoint must have component_name='chat_generator'.
+        - ToolBreakpoint must have component_name='tool_invoker'.
+    """
 
-        self.agent_name = agent_name
+    agent_name: str
+    break_point: Union[Breakpoint, ToolBreakpoint]
 
+    def __post_init__(self):
```

Comment:
I updated this to be a `__post_init__` so we could just focus on running the validation checks.

Reply:
nice 👍🏽